### PR TITLE
[chainerx] add `floor`

### PR DESCRIPTION
**Pull request by kshitij12345**
Add `floor` to ChainerX

References #6423

### CODE REVIEW

**Comment by hvy** [comment]
For routines that are not backpropable, you can simply skip the `BackwardBuilder` step. FYI, [This PR](https://github.com/chainer/chainer/pull/6621/files#diff-2c0c90610b0e5cc7cf41dfd227e5dbe6R47) will soon change how such routines are written so that disconnected graphs will become backpropable too.

**Comment by kshitij12345** [comment]
* Have rebased.
* Skipped BackwardBuilder
* Updated Test for only checking for Equality of Output with Numpy

**Comment by kshitij12345** [comment]
Have used `REGISTER_OP` macro in the latest commit. Will remove it if that is not the current direction of moving forward.

**Comment by hvy** [comment]
Great, you noticed the op registration too. We however changes the design a bit. Could you apply some changes similar to this PR? https://github.com/chainer/chainer/pull/6744

**Comment by kshitij12345** [comment]
@hvy have updated the Op Calls please review

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 1]
```diff
@@ -1893,6 +1893,30 @@ def func(self, xp, a):
         return xp.arctan(a)
 
 
+@chainerx.testing.numpy_chainerx_array_equal(strides_check=False)
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Ditto of https://github.com/chainer/chainer/pull/6705#discussion_r272822229

**Comment by hvy** [comment]
Jenkins, test this please.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 1 above)
Comment (kshitij12345) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: @hvy Thanks for pointing out. On further inspection it is not required as we manually cast `numpy` output to expected `chainerx` outputs. This is done as `numpy` defaults to `float64` while `chainerx` to `float32`.

Have removed the `strides_check=False`.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
The changes look good but it's conflicting now that `ceil` has been merged. Could you fix the conflicts?

**Comment by kshitij12345** [comment]
Have fixed the conflicts and updated the test to share the `params`. Please review.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM, let me check the CI.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Removing `to-be-backported` as we decided not to backport non-critical PRs.

**Pull request merged**